core/types/governance_new_market.go

```diff
@@ -431,6 +433,14 @@ func (i InstrumentConfigurationFuture) Assets() []string {
 	return i.Future.Assets()
 }
 
+func (_ InstrumentConfigurationFuture) IsFuture() bool {
+	return true
```

Comment:
that doesn't feel right. Why not have a product enum instead?

Reply:
Agreed, will add something akin to what we have on the proposal type (getTermType I believe), along with getters for specific types. That should cut down significantly on this kind of interface bloat